engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/event/EndEventProcessor.java

```diff
@@ -324,10 +328,23 @@ public boolean isSuitableForEvent(final ExecutableEndEvent element) {
 
     @Override
     public void onActivate(final ExecutableEndEvent element, final BpmnElementContext activating) {
+      compensationSubscriptionBehaviour.updateCompensationSubscription(activating);
```

Comment:
:wrench: The subscription `TRIGGERED` event is part of the subscription lifecycle. We should write one event per subscription. If there are no subscriptions created, we should not write an event. 

We could move the call directly above the activation of the compensation handlers.